tests/test_garbage_collect.py

```diff
@@ -30,15 +27,9 @@ def _get_time_created(d=0, h=0, s=0):
 
 
 def _get_fake_controller(*args):
-    @attr.s(kw_only=True, frozen=True)
-    class TestRepo(Repository):
-        is_temp_repo = pp_attr.pulp_attrib(
-            default=None, type=str, pulp_field="notes.pub_temp_repo"
-        )
-
     controller = FakeController()
     for repo in args:
-        yum_repo = TestRepo.from_data(repo)
+        yum_repo = Repository.from_data(repo)
```

Comment:
Really I don't think you should need to use from_data at all.  Can you just create the repository objects with the attributes you need?  It's intended to be used that way, so converting from raw Pulp data happens in a bare minimum of places.

You would probably have to help me get https://github.com/release-engineering/pubtools-pulplib/pull/36 done and released first though.

Reply:
Updated the tests to use created repo objects using Repository attributes